cmd/auth-handler.go

```diff
@@ -101,6 +102,25 @@ func getRequestAuthType(r *http.Request) authType {
 	return authTypeUnknown
 }
 
+// checkAdminRequestAuthType checks whether the request is a valid signature V2 or V4 request.
+// It does not accept presigned or JWT or anonymous requests.
+func checkAdminRequestAuthType(r *http.Request, region string) APIErrorCode {
+	var s3Err APIErrorCode
+	switch getRequestAuthType(r) {
+	default:
+		s3Err = ErrAccessDenied // default to access denied
+	case authTypeSignedV2:
+		s3Err = isReqAuthenticatedV2(r)
```

Comment:
Would it make sense to disable V2 authentication completely for admin API?

Reply:
:+1:

Reply:
It would be better to disable V2 for admin-API and quickly take this patch in and make a release

Reply:
I wanted to separate a breaking change from a security fix. Users may refuse updating because it breaks their system... 
(Hopefully nobody found this bug **and** uses it as feature!)
However if we agree that we remove V2 support **now** - I'm okay with that.

Reply:
@aead just discussed with AB, we can remove V2 support for admin API.